docs-source/examples/three_legged_oauth.rst

```diff
@@ -0,0 +1,191 @@
+Three Legged OAuth with Flask
+-----------------------------
+
+This type of authorization is used for web login with a server-side
+application. For example, a Django app or other application server handles
+requests.
+
+This example uses Flask, but should be easily portable to other application
+frameworks.
+
+
+Components
+~~~~~~~~~~
+
+There are two components to this application: login and logout.
+
+Login sends a user to Globus Auth to get credentials, and then may act on the
+user's behalf.
+Logout invalidates server-side credentials, so that the application may no
+longer take actions for the user, and the client-side session,
+allowing for a fresh login if desired.
+
+Register an App
+~~~~~~~~~~~~~~~
+
+Go to developers.globus.org and register a new application.
+Make sure that the "Native App" checkbox is unchecked, and list
+``http://localhost:5000/login`` in the "Redirect URIs".
+
+Set the Scopes to ``openid``, ``profile``, ``email``,
+``urn:globus:auth:scope:transfer.api.globus.org:all``.
+
+On the projects page, expand the client description and click "Generate
+Secret".
+Save the resulting secret a file named ``example_app.conf``, along with the client ID:
+
+.. code-block:: python
+
+    SERVER_NAME = 'localhost:5000'
+    # this is the session secret, used to protect the Flask session. You should
+    # use a longer secret string known only to your application
+    # details are beyond the scope of this example
+    SECRET_KEY = 'abc123!'
+
+    APP_CLIENT_ID = '<CLIENT_ID>'
+    APP_CLIENT_SECRET = '<CLIENT_SECRET>'
+
+Shared Utilities
+~~~~~~~~~~~~~~~~
+
+Some pieces that are of use for both parts of this flow.
+
+First, you'll need to install ``Flask`` and the ``globus-sdk``.
+Assuming you want to do so into a fresh virtualenv:
+
+.. code-block:: bash
+
+    $ virtualenv example-venv
+    ...
+    $ source example-venv/bin/activate
+    $ pip install Flask==0.11.1 globus-sdk
+    ...
+
+You'll also want a shared function for loading the SDK ``AuthClient`` which
+represents your application, as you'll need it in a couple of places. Create
+it, along with the defintiion for your Flask app, in ``example_app.py``:
+
+.. code-block:: python
+
+    from flask import Flask, url_for, session
+    import globus_sdk
+
+    app = Flask(__name__)
+    app.config.from_pyfile('example_app.conf')
+
+    def load_app_client():
+        return globus_sdk.ConfidentialAppAuthClient(
+            app.config['APP_CLIENT_ID'], app.config['APP_CLIENT_SECRET'])
+
```

Comment:
I'm not entirely sure this is the best place, but the example code needs to run the app somewhere.
`if __name__ == "__main__":`
    `app.run()`